### ISSUE TITLE
Deploy with npm to Heroku

### ISSUE THREAD

**Issue by Plavit** [Military-Tactical-Graphics/milsymbol-generator#6]
GH pages can only host static html pages it turns out, replaces #3

**Comment by Plavit** [comment]
Crashing so far, here is possible troubleshooting:
https://dev.to/lawrence_eagles/causes-of-heroku-h10-app-crashed-error-and-how-to-solve-them-3jnl

**Comment by Plavit** [comment]
I made some progress, here is the fail statur right now:

`2021-01-03T18:22:10.000000+00:00 app[api]: Build started by user marek.szeles@gmail.com
        2021-01-03T18:22:52.648938+00:00 app[api]: Release v7 created by user marek.szeles@gmail.com
        2021-01-03T18:22:52.648938+00:00 app[api]: Deploy 12ad6030 by user marek.szeles@gmail.com
        2021-01-03T18:22:53.793610+00:00 heroku[web.1]: State changed from crashed to starting
        2021-01-03T18:22:53.000000+00:00 app[api]: Build succeeded
        2021-01-03T18:22:59.517609+00:00 heroku[web.1]: Starting process with command `npm start`
        2021-01-03T18:23:02.702400+00:00 app[web.1]: 
        2021-01-03T18:23:02.702460+00:00 app[web.1]: > milsymbol-generator@1.0.0 start /app
        2021-01-03T18:23:02.702461+00:00 app[web.1]: > node src/index.js
        2021-01-03T18:23:02.702461+00:00 app[web.1]: 
        2021-01-03T18:23:02.756821+00:00 app[web.1]: (node:23) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
        2021-01-03T18:23:02.759254+00:00 app[web.1]: /app/src/index.js:2
        2021-01-03T18:23:02.759264+00:00 app[web.1]: import mdc from "material-components-web/dist/material-components-web.min.js";
        2021-01-03T18:23:02.759264+00:00 app[web.1]: ^^^^^^
        2021-01-03T18:23:02.759265+00:00 app[web.1]: 
        2021-01-03T18:23:02.759265+00:00 app[web.1]: SyntaxError: Cannot use import statement outside a module
        2021-01-03T18:23:02.759266+00:00 app[web.1]:     at wrapSafe (internal/modules/cjs/loader.js:915:16)
        2021-01-03T18:23:02.759266+00:00 app[web.1]:     at Module._compile (internal/modules/cjs/loader.js:963:27)
        2021-01-03T18:23:02.759266+00:00 app[web.1]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
        2021-01-03T18:23:02.759267+00:00 app[web.1]:     at Module.load (internal/modules/cjs/loader.js:863:32)
        2021-01-03T18:23:02.759267+00:00 app[web.1]:     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
        2021-01-03T18:23:02.759276+00:00 app[web.1]:     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
        2021-01-03T18:23:02.759277+00:00 app[web.1]:     at internal/main/run_main_module.js:17:47
        2021-01-03T18:23:02.769650+00:00 app[web.1]: npm ERR! code ELIFECYCLE
        2021-01-03T18:23:02.769886+00:00 app[web.1]: npm ERR! errno 1
        2021-01-03T18:23:02.774699+00:00 app[web.1]: npm ERR! milsymbol-generator@1.0.0 start: `node src/index.js`
        2021-01-03T18:23:02.774791+00:00 app[web.1]: npm ERR! Exit status 1
        2021-01-03T18:23:02.774968+00:00 app[web.1]: npm ERR! 
        2021-01-03T18:23:02.775056+00:00 app[web.1]: npm ERR! Failed at the milsymbol-generator@1.0.0 start script.
        2021-01-03T18:23:02.775157+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
        2021-01-03T18:23:03.768974+00:00 app[web.1]: 
        2021-01-03T18:23:03.769211+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
        2021-01-03T18:23:03.769362+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2021-01-03T18_23_02_775Z-debug.log
        2021-01-03T18:23:03.825499+00:00 heroku[web.1]: Process exited with status 1
        2021-01-03T18:23:03.867982+00:00 heroku[web.1]: State changed from starting to crashed
        2021-01-03T18:23:03.872162+00:00 heroku[web.1]: State changed from crashed to starting
        2021-01-03T18:23:07.963770+00:00 heroku[web.1]: Starting process with command `npm start`
        2021-01-03T18:23:12.113595+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
        2021-01-03T18:23:12.113934+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2021-01-03T18_23_12_079Z-debug.log
        2021-01-03T18:23:12.195919+00:00 heroku[web.1]: Process exited with status 1
        2021-01-03T18:23:11.853976+00:00 app[web.1]: 
        2021-01-03T18:23:11.853994+00:00 app[web.1]: > milsymbol-generator@1.0.0 start /app
        2021-01-03T18:23:11.853995+00:00 app[web.1]: > node src/index.js
        2021-01-03T18:23:11.853995+00:00 app[web.1]: 
        2021-01-03T18:23:12.019711+00:00 app[web.1]: (node:23) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
        2021-01-03T18:23:12.027360+00:00 app[web.1]: /app/src/index.js:2
        2021-01-03T18:23:12.027361+00:00 app[web.1]: import mdc from "material-components-web/dist/material-components-web.min.js";
        2021-01-03T18:23:12.027362+00:00 app[web.1]: ^^^^^^
        2021-01-03T18:23:12.027362+00:00 app[web.1]: 
        2021-01-03T18:23:12.027363+00:00 app[web.1]: SyntaxError: Cannot use import statement outside a module
        2021-01-03T18:23:12.027363+00:00 app[web.1]:     at wrapSafe (internal/modules/cjs/loader.js:915:16)
        2021-01-03T18:23:12.027364+00:00 app[web.1]:     at Module._compile (internal/modules/cjs/loader.js:963:27)
        2021-01-03T18:23:12.027364+00:00 app[web.1]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
        2021-01-03T18:23:12.027364+00:00 app[web.1]:     at Module.load (internal/modules/cjs/loader.js:863:32)
        2021-01-03T18:23:12.027365+00:00 app[web.1]:     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
        2021-01-03T18:23:12.027365+00:00 app[web.1]:     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
        2021-01-03T18:23:12.027366+00:00 app[web.1]:     at internal/main/run_main_module.js:17:47
        2021-01-03T18:23:12.060000+00:00 app[web.1]: npm ERR! code ELIFECYCLE
        2021-01-03T18:23:12.060508+00:00 app[web.1]: npm ERR! errno 1
        2021-01-03T18:23:12.077471+00:00 app[web.1]: npm ERR! milsymbol-generator@1.0.0 start: `node src/index.js`
        2021-01-03T18:23:12.077736+00:00 app[web.1]: npm ERR! Exit status 1
        2021-01-03T18:23:12.078035+00:00 app[web.1]: npm ERR! 
        2021-01-03T18:23:12.078308+00:00 app[web.1]: npm ERR! Failed at the milsymbol-generator@1.0.0 start script.
        2021-01-03T18:23:12.078580+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
        2021-01-03T18:23:12.099604+00:00 app[web.1]: 
        2021-01-03T18:23:12.372616+00:00 heroku[web.1]: State changed from starting to crashed
        2021-01-03T18:24:06.150025+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=milgraphics.herokuapp.com request_id=e888d64e-2e30-4b2d-8f74-ae7f0b99a294 fwd="89.233.185.226" dyno= connect= service= status=503 bytes= protocol=https
        2021-01-03T18:24:06.742131+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=milgraphics.herokuapp.com request_id=5e8d495d-bf69-4d4c-a2d9-f45bad6d707c fwd="89.233.185.226" dyno= connect= service= status=503 bytes= protocol=https`

**Comment by Plavit** [comment]
Seems like the start script was referencing a wrong file... but still failing. To be continued...

**Comment by Plavit** [comment]
Error not very verbose now:
```
2021-01-04T13:32:45.343751+00:00 heroku[web.1]: Starting process with command `npm start`
        2021-01-04T13:32:49.021557+00:00 app[web.1]: 
        2021-01-04T13:32:49.021586+00:00 app[web.1]: > milsymbol-generator@1.0.0 start /app
        2021-01-04T13:32:49.021587+00:00 app[web.1]: > node unitgenerator/milsymbol-unit-generator.js
        2021-01-04T13:32:49.021587+00:00 app[web.1]: 
        2021-01-04T13:32:49.421201+00:00 app[web.1]: milsymbol.js 2.0.0 - Copyright (c) 2017 Måns Beckman  http://www.spatialillusions.com
        2021-01-04T13:32:49.529445+00:00 heroku[web.1]: Process exited with status 0
        2021-01-04T13:32:49.584481+00:00 heroku[web.1]: State changed from starting to crashed
```

**Comment by Plavit** [comment]
Resolved by be95ce0, basically I had to create a separate node server to deploy the static index.html but simulate client-side javascript, all the while playing by Heroku rules. It was fun.
